sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c

```diff
@@ -0,0 +1,447 @@
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralAonTimerAon) {
+    uint32_t irq =
+        (irq_id - (dif_rv_plic_irq_id_t)
+                      kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+    // We should not get aon timer interrupts since escalation suppresses them.
+    CHECK(false, "Unexpected aon timer interrupt %d", irq);
+  } else if (peripheral == kTopEarlgreyPlicPeripheralAlertHandler) {
+    CHECK(irq_id == kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+          "Unexpected irq_id, expected %d, got %d",
+          kTopEarlgreyPlicIrqIdAlertHandlerClassa, irq_id);
+  }
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
+  // register.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kPlicTarget, irq_id));
+  LOG_INFO("Regular external ISR exiting");
+}
+
+void ottf_load_integrity_error_handler(void) {
+  dif_rv_plic_irq_id_t irq_id;
+
+  LOG_INFO("At load integrity error handler");
+
+  uint32_t mcause = ibex_mcause_read();
+  CHECK(mcause == kIbexInternalIrqLoadInteg,
+        "Unexpected mcause: expected 0x%x, got 0x%x", kIbexInternalIrqLoadInteg,
+        mcause);
+  uint32_t mtval = ibex_mtval_read();
+  CHECK(mtval == kErrorRamAddress, "Unexpected mtval: expected 0x%x, got 0x%x",
+        kErrorRamAddress, mtval);
+
+  rv_core_ibex_fault_checker(true);
+
+  // There may be multiple interrupts due to the alert firing, so this keeps an
+  // interrupt counter and errors-out if there are too many interrupts.
+
+  // Increment the interrupt count and detect overflows.
+  uint32_t interrupt_count =
+      flash_ctrl_testutils_counter_get(kCounterInterrupt);
+  if (interrupt_count > kMaxInterrupts) {
+    CHECK(false, "Reset count %d got too many interrupts (%d)", reset_count,
+          interrupt_count);
+  }
+  flash_ctrl_testutils_counter_set_at_least(
+      &flash_ctrl_state, kCounterInterrupt, interrupt_count + 1);
+
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+
+  LOG_INFO("Got irq_id 0x%x (%d)", irq_id, irq_id);
+
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  if (peripheral == kTopEarlgreyPlicPeripheralAonTimerAon) {
+    uint32_t irq =
+        (irq_id - (dif_rv_plic_irq_id_t)
+                      kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+    // We should not get aon timer interrupts since escalation suppresses them.
+    CHECK(false, "Unexpected aon timer interrupt %d", irq);
+  } else if (peripheral == kTopEarlgreyPlicPeripheralAlertHandler) {
+    LOG_INFO("Got expected alert handler interrupt %d", irq_id);
+
+    // Disable this interrupt to prevent it from continuously firing. This
+    // should not prevent escalation to ensue.
+    CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(&plic, irq_id, kPlicTarget,
+                                             kDifToggleEnabled));
+  }
```

Comment:
Yeah I figured it would, I suppose it is a matter of preference/convention but I would've expected the alert handler IRQ to be serviced (i.e., disabling further IRQs from the alert handler at the PLIC after the first one as you do) in the external ISR rather than in a (synchronous) exception  handler,  since alert IRQs are really an asynchronous exception, since an alert can be triggered by an asynchronous event like a bit flip. It just so happens that this alert is synchronous since it happens on an instruction fetch, at the same time the internal `load_integrity_error` exception is fired, no? It is really just a matter of preference/convention as I said, and given this test configures things in somewhat odd ways to test specific data integrity features of the chip, I am ok with leaving as is.